### PR TITLE
 The "year" denomination is deprecated

### LINKED ISSUE #3606: Deprecate `years` suffix
The `years` literal number suffix is defined as 31536000 seconds at https://github.com/ethereum/solidity/blob/develop/libsolidity/ast/Types.cpp#L747

This is incorrect - the correct value is 31557600 seconds.

A true Earth year is closer to [365.256363004 days](https://en.wikipedia.org/wiki/Earth) which is standardized as 365.25 days of 86400 SI seconds each in the [Julian Year](https://en.wikipedia.org/wiki/Year), as used in astronomy and other sciences.

However, as civilians and the Gregorian calendar uses [common](https://en.wikipedia.org/wiki/Common_year) and [leap years](https://en.wikipedia.org/wiki/Leap_year), it may be worthwhile to consider deprecating `years` to avoid potential confusion altogether.

### PR DESCRIPTION
**Pull request by axic**
Fixes #3606.

### CODE REVIEW

File: test/libsolidity/syntaxTests/denominations/denominations.sol [hunk 1]
```diff
@@ -0,0 +1,10 @@
+contract C {
```
Comment (chriseth) [inline test/libsolidity/syntaxTests/denominations/denominations.sol:1]: Is this a new test? If yes, please remove the error. If no, please remove the original :)

Comment (axic) [reply test/libsolidity/syntaxTests/denominations/denominations.sol:1]: Added the error on purpose to have more things to check for. You want it to be in a separate file?

Comment (chriseth) [reply test/libsolidity/syntaxTests/denominations/denominations.sol:1]: Yes, I think that would be better.

Comment (axic) [reply test/libsolidity/syntaxTests/denominations/denominations.sol:1]: I've split it.

**Review by chriseth — APPROVED** [review]

**Pull request merged**